migrator.go

```diff
@@ -47,6 +47,78 @@ func (m Migrator) FullDataTypeOf(field *schema.Field) clause.Expr {
 	return expr
 }
 
+// MigrateColumnUnique migrate column's UNIQUE constraint.
+// In MySQL, ColumnType's Unique is affected by UniqueIndex, so we have to take care of the UniqueIndex.
+func (m Migrator) MigrateColumnUnique(value interface{}, field *schema.Field, columnType gorm.ColumnType) error {
+	unique, ok := columnType.Unique()
+	if !ok || field.PrimaryKey {
+		return nil // skip primary key
+	}
+
+	queryTx, execTx := m.GetQueryAndExecTx()
+	return m.RunWithValue(value, func(stmt *gorm.Statement) error {
+		// We're currently only receiving boolean values on `Unique` tag,
+		// so the UniqueConstraint name is fixed
+		constraint := m.DB.NamingStrategy.UniqueName(stmt.Table, field.DBName)
+		if unique {
+			// Clean up redundant unique indexes
+			indexes, _ := queryTx.Migrator().GetIndexes(value)
+			for _, index := range indexes {
+				if uni, ok := index.Unique(); !ok || !uni {
+					continue
+				}
+				if columns := index.Columns(); len(columns) != 1 || columns[0] != field.DBName {
+					continue
+				}
+				if name := index.Name(); name == constraint || name == field.UniqueIndex {
+					continue
+				}
+				if err := execTx.Migrator().DropIndex(value, index.Name()); err != nil {
```

Comment:
I had a column for which I changed from `index` to `uniqueIndex`

MyColumn       int32 `gorm:"index"` 
Gorm had created idx_<table name>_<column_name>

which I changed to 

MyColumn       int32 `gorm:"uniqueIndex"`
Gorm retained old index idx_<table name>_<column_name> and created a new unique index `<column_name>`

After this updatae

gorm.io/driver/mysql v1.5.2 => v1.5.6
gorm.io/gorm v1.25.5 => v1.25.10

It dropped index <column_name> which had unique index and tried to create index with name idx_<table name>_<column_name> and failed.

Ideally we should have also dropped idx_<table name>_<column_name> once we realised we are creating it.

As per the text here - https://gorm.io/docs/migration.html 'It WON’T delete unused columns to protect your data.' we should not have done this right '// Clean up redundant unique indexes'

Option could have been 
1. Check if can't create - stop 
2. Drop what we have figured we want to create?

Please let me know.